prow/README.md

```diff
@@ -4,206 +4,383 @@ Metal3 Prow dashboard: <https://prow.apps.test.metal3.io>
+If there is an existing instance, you most likely just have to take the credentials
+from the secure place and generate the files below from them.
+
+Please set the following environment variables with the relevant credentials.
+Then you will be able to just copy and paste the snippets below.
+
+- `APP_CRED_ID`
+- `APP_CRED_SECRET`
+- `S3_ACCESS_KEY`
+- `S3_SECRET_KEY`
+- `HMAC_TOKEN`
+- `GITHUB_TOKEN`
+- `CHERRYPICK_TOKEN`
+
+Now you are ready to create the files.
+
+1. Create `clouds.yaml` secret used by the CAPO controllers to manage the infrastructure.
+
+   ```bash
+   cat > capo-cluster/clouds.yaml <<-EOF
+   clouds:
+     prow:
+       auth:
+         auth_url: https://kna1.citycloud.com:5000/v3/
+         auth_type: v3applicationcredential
+         application_credential_id: ${APP_CRED_ID}
+         application_credential_secret: ${APP_CRED_SECRET}
+         domain_name: CCP_Domain_37137
+         project_name: Default Project 37137
+       region_name: Kna1
+       verify: false
+   EOF
    ```
 
-1. Create a secret with the GCS credentials
+1. Create `cloud.conf` secret for the cloud provider integration.
+
+   ```bash
+   cat > cluster-resources/cloud.conf <<-EOF
+   [Global]
+   auth-url=https://kna1.citycloud.com:5000/v3/
+   application-credential-id=${APP_CRED_ID}
+   application-credential-secret=${APP_CRED_SECRET}
+   region=Kna1
+   domain-name=CCP_Domain_37137
+   EOF
+   ```
 
-   ```shell
-   kubectl create secret generic gcs-credentials \
-       --from-file=service-account.json=service-account.json
+1. Get the kubeconfig and save it as `capo-cluster/kubeconfig.yaml`.
+
+1. Create S3 credentials secret.
+
+   ```bash
+   # Create service-account.json
+   cat > manifests/overlays/metal3/service-account.json <<-EOF
+   {
+      "region": "Kna1",
+      "access_key": "${S3_ACCESS_KEY}",
+      "endpoint": "s3-kna1.citycloud.com:8080",
+      "insecure": false,
+      "s3_force_path_style": true,
+      "secret_key": "${S3_SECRET_KEY}"
+   }
+   EOF
    ```
 
-1. While applying prow manifests, we are going to create an Ingress object to
-   redirect traffic to the deck and hook services. We are installing
-   [NGINX Ingress controller](https://kubernetes.github.io/ingress-nginx/deploy/)
-   with helm, but you can install your preferred one.
+1. Create configuration file for `s3cmd`.
 
-   ```shell
-   helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
-   helm install ingress-nginx/ingress-nginx \
-       --set controller.hostNetwork=true,controller.service.type=""\
-   ,controller.kind=DaemonSet --generate-name
-   helm repo update
+   ```bash
+   s3cmd --config .s3cfg --configure
+   # Provide Access key and Secret key from the openstack credentials created above.
+   # Leave Default Region as US.
+   # Set S3 Endpoint to s3-kna1.citycloud.com:8080
+   # Set DNS-style bucket+hostname:port template to the same s3-kna1.citycloud.com:8080
+   # Default values for the rest. And save settings.
```

Comment:
This section needs a bit more explanation. What is this used for?

Reply:
Good point! I'll add that it is used for creating the buckets.

Reply:
If it is ok, I could do this in a follow up where I also remove the extra copies of config that is now there just to make the config-validator happy.

Reply:
Yeah thats fine.

Reply:
Added it now when I anyway got more comments to address :slightly_smiling_face: